aws/src/main/java/org/apache/iceberg/aws/glue/GlueCatalog.java

```diff
@@ -110,13 +110,15 @@ public GlueCatalog() {
   @Override
   public void initialize(String name, Map<String, String> properties) {
     AwsClientFactory awsClientFactory;
+    ImmutableMap.Builder<String, String> builder = ImmutableMap.<String, String>builder().putAll(properties);
```

Comment:
I think we can make this a `tableSpecificPropertiesBuilder`, and at the end if this builds an empty map it means there is no custom table specific catalog properties, and we should use the else clause.

Reply:
[Doubt] If the user passes only the table level tag properties, then we will build the map and according to this logic we would skip the else block. Thereby not initialising `awsClientFactory` and `catalogFileIO` which should not be the intended behaviour.